infra/pr_helper.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Adds comments for PR to provide more information for approvers."""
+import base64
+import json
+import os
+import subprocess
+
+import requests
+import yaml
+
+OWNER = 'google'
+REPO = 'oss-fuzz'
+GITHUB_URL = 'https://github.com/'
+API_URL = 'https://api.github.com'
+BASE_URL = f'{API_URL}/repos/{OWNER}/{REPO}'
+BRANCH = 'master'
+
+
+def get_projects_path(pr_number, headers):
+  """Gets the current project path."""
+  response = requests.get(f'{BASE_URL}/pulls/{pr_number}/files',
+                          headers=headers)
+
+  projects_path = set()
+  for file in response.json():
+    file_path = file['filename']
+    dir_path = os.path.dirname(file_path)
+    if 'projects' in dir_path:
+      projects_path.add(dir_path)
+  return list(projects_path)
+
+
+def get_author_email(pr_number, headers):
+  """Retrieves the author's email address for a pull request,
+  including non-public emails."""
+  commits_response = requests.get(f'{BASE_URL}/pulls/{pr_number}/commits',
+                                  headers=headers)
+  if not commits_response.ok:
+    return None
+  email = commits_response.json()[0]['commit']['author']['email']
```

Comment:
Can we check that this is verified by github? Commit authors can be forged easily, and we need to make sure this email is coming from a verified email on the github account.

Reply:
I tried to use [Github email api](https://docs.github.com/en/rest/users/emails?apiVersion=2022-11-28#list-email-addresses-for-the-authenticated-user) first, but it can only obtain user's public email. For those who set up their email as private on github account (like myself), we can't obtain their verified emails by this way. So I used commit information instead. Please let me know if there is other approaches.

Reply:
There is also this "verification" field in the response (https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#get-a-commit) that we could leverage, but t his may not be set in all cases. 

If we can't verify either the email from the email API, nor from the verification field, then we should consider the email to be completely untrusted and forgeable and add this to our PR message. 

The other cases (checking for maintainer, or previous contributor), can be done by GitHub username right? Those we can continue to trust.

Reply:
Yes, the maintainer and previous contributor checks can be done by username. The email is only use to compare with the contact emails on project.yaml. Added `verification` field check. If the email is not verified, it will leave a message and continue previous contributor check.